File: backend/models/application.go
package models

import "time"

type ApplicationStatus uint8

const (
	StatusApplied ApplicationStatus = iota
	StatusOAReceived
	StatusInterviewing
	StatusAccepted
	StatusRejected
)

func (s ApplicationStatus) String() string {
	switch s {
	case StatusApplied:
		return "Applied"
	case StatusOAReceived:
		return "OA Received"
	case StatusInterviewing:
		return "Interviewing"
	case StatusAccepted:
		return "Accepted"
	case StatusRejected:
		return "Rejected"
	default:
		return "Unknown"
	}
}

type Application struct {
	ID          uint              `gorm:"primaryKey" json:"id"`
	Company     string            `json:"company"`
	Position    string            `json:"position"`
	Status      ApplicationStatus `json:"status"` // e.g., "Applied", "Interview", etc.
	Location    string            `json:"location"`
	AppliedDate time.Time         `json:"applied_date"`
	Term        string            `json:"term"`                            // e.g., "Summer 2025"
	Note        string            `gorm:"size:1048" json:"note,omitempty"` // Optional note, up to 1048 chars
	ResumeURL   string            `json:"resume_url"`
	UserID      uint              `json:"user_id"`                                 // Set automatically by server
	User        User              `gorm:"foreignKey:UserID" json:"user,omitempty"` // Only in responses
}

File: frontend/src/lib/types/application.ts
export interface Application {
    id: number;
    company: string;
    position: string;
    status: ApplicationStatus;
    location: string;
    applied_date: string;
    term: string;
    note?: string;
    resume_url: string;
    user_id: number;
}

export enum ApplicationStatus {
    Applied = 0,
    OAReceived = 1,
    Interviewing = 2,
    Accepted = 3,
    Rejected = 4
}

export interface LoadingState {
    isLoading: boolean;
    error: string | null;
}

export interface ApiResponse<T = any> {
    data?: T;
    error?: string;
    success?: boolean;
}

export const STATUS_COLORS = {
    [ApplicationStatus.Applied]: 'primary',
    [ApplicationStatus.OAReceived]: 'info',
    [ApplicationStatus.Interviewing]: 'warning',
    [ApplicationStatus.Accepted]: 'success',
    [ApplicationStatus.Rejected]: 'danger'
};
